src/main.rs

```diff
@@ -306,6 +326,52 @@ impl Fixer {
         }
     }
 
+    fn build_file_info_breakpad(bin_file: &str, syms_dir: &str) -> Result<FileInfo, String> {
+        // We must find the `.sym` file for this `bin_file`, as produced by the
+        // Firefox build system. A running example:
+        // - `bin_file` is `bin/libxul.so`
+        // - `syms_dir` is `syms/`
+        // - The symbols for `libxul.so` are in `syms/libxul.so/<uuid>/libxul.so.sym`
+        let bin_file = Path::new(bin_file);
+
+        // - `pathless_bin_file` is `libxul.so`
+        let pathless_bin_file = bin_file.file_name().ok_or("read breakpad symbols for")?;
+
+        // - `pathless_sym_file` is `libxul.so.sym`
+        let mut pathless_sym_file = pathless_bin_file.to_str().unwrap().to_string();
+        pathless_sym_file.push_str(".sym");
+
+        // - `syms_bin_dir` is `syms/libxul.so/`
+        let mut syms_bin_dir = PathBuf::new();
+        syms_bin_dir.push(&syms_dir);
+        syms_bin_dir.push(&pathless_bin_file);
+
+        // - `syms_bin_entries` iterates over `syms/libxul.so/`
+        let mut syms_bin_entries = fs::read_dir(&syms_bin_dir)
+            .map_err(|_| format!("read breakpad symbols dir `{}` for", syms_bin_dir.display()))?;
+
+        // - `syms_uuid_dir` is `syms/libxul.so/<uuid>/`
+        let syms_uuid_dir = if let (Some(d), None) =
+            (syms_bin_entries.next(), syms_bin_entries.next())
+        {
+            d.map_err(|_| format!("read breakpad symbols dir `{}` for", syms_bin_dir.display()))?
+                .path()
+        } else {
+            return Err(format!(
+                "read breakpad symbols dir `{}` (not exactly one subdir) for",
+                syms_bin_dir.display()
```

Comment:
There are scenarios where we have more than one <uuid> dir under the same object dir. The most typical being libxul.so in automation builds where you always have two subdirs, the one for "regular" libxul and the gtest-enabled libxul. Naturally the script has no way to tell which is which (it would need to access the real libxul.so to figure out the uuid) so we'll have to enforce this constraint when we invoke the script. Have you thought how to deal with that?

Reply:
The existing script has logic to deal with this: https://searchfox.org/mozilla-central/source/tools/rb/fix_stack_using_bpsyms.py#122-139,155-156. It invokes a `fileid` executable to get the UUID.

I didn't copy this logic because it didn't seem necessary, but sounds like it will be on automation. Thanks for the info!

Reply:
Aha, nice catch!